tests/unit/timers_test.ts

```diff
@@ -767,3 +767,15 @@ Deno.test({
     assert(result >= 1000);
   },
 });
+
+// Regression test for https://github.com/denoland/deno/issues/20663
+Deno.test({
+  name: "regression for #20663",
+  fn: async () => {
+    const signal = AbortSignal.timeout(2000);
+    const t = setTimeout(() => {
```

Comment:
To properly test this fix, I believe you should create an AbortSignal.timeout and let it just leak out of the test.